okteto/okteto
# DEV-258: Changed command used to get the latest commit for a folder

# Proposed changes

Fixes DEV-258

As the installer is cloning repositories always using only one commit, the build context calculation doesn't work well because you have different behaviors from local and the installer, generating unnecessary builds. As cloning the whole history might impact performance, we are changing the CLI to not use the latest commit of the folder but instead, use a hash of the files tracked by git within that folder. 

The command to be used is `/usr/bin/git --no-optional-locks ls-files -s . | /usr/bin/git --no-optional-locks hash-object --stdin`

## How to validate

1. Verify general behavior of Smart Builds. Check that cache is being use correctly when it has to

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

## Tests done

* I tested that a simple commit in a repo without affecting to none of the build context don't trigger a build of any image in the installer if images were already build
* I tested that a change in a single build context only triggers the rebuild of that image within the installer
* I was testing that it works on windows machines
* I was testing also that locally, smart builds keeps working as expected
